mlforecast/forecast.py

```diff
@@ -233,7 +238,30 @@ def predict(
             raise ValueError(
                 "No fitted models found. You have to call fit or preprocess + fit_models."
             )
-        return self.ts.predict(
+
+        if new_data is not None:
+            new_ts = TimeSeries(
+                self.ts.freq,
+                self.ts.lags,
+                self.ts.lag_transforms,
+                self.ts.date_features,
+                self.ts.differences,
+                self.ts.num_threads,
+            )
+            new_ts.fit_transform(
```

Comment:
The transform part isn't necessary. We can probably move the checks that are done inside fit_transform before calling _fit into _fit, if that's what you're worried about.
```suggestion
            new_ts._fit(
```

Reply:
fixed